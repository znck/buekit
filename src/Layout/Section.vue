<script>
import PropTypes from '@znck/prop-types'
import { style, styleResolver } from '../utils'
import { sizes } from '../shared'
import { createTag } from '../mixins'

export default {
  name: 'Section',
  functional: true,
  mixins: [createTag('section')],
  props: { size: PropTypes.oneOf(sizes) },
  render (h, ctx) {
    const _ = styleResolver(ctx.$style)
    const { size, tag } = ctx.props

    return style(
      h(tag, ctx.data, ctx.children),
      _('section'),
      size && _(`is-${size}`)
    )
  }
}
</script>

<style lang="scss" module>
@import '~bulma/sass/utilities/_all.sass';
@import '../_base.scss';
@import '~bulma/sass/layout/section.sass';
</style>

