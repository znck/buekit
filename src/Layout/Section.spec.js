import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'

import Section from './Section.vue'

describe('Section.vue', () => {
	it('matches snapshot', () => {
		const wrapper = shallow(Section, { context: { props: {} } })
		const renderer = createRenderer()

		renderer.renderToString(wrapper.vm, (err, str) => {
			if (err) {
				throw new Error(err)
			}
			expect(str).toMatchSnapshot()
		})
	})
})
