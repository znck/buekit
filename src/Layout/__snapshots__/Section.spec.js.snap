// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section.vue matches snapshot 1`] = `<section class="section is-undefined"></section>`;
